static/js/stream_data.js

```diff
@@ -345,6 +345,17 @@ exports.add_subscriber = function (stream_name, user_id) {
     return true;
 };
 
+exports.remove_deactivated_user_from_all_streams = function (user) {
+    (stream_info.values()).forEach(function (stream) {
+        if (exports.user_is_subscribed(stream.name, user.email)) {
+            exports.remove_subscriber(stream.name, user.user_id);
+            var sub = exports.get_sub(stream.name);
+            stream_edit.rerender_subscribers_list(sub);
```

Comment:
This is really expensive; I think we only need to rerender the subscriber list for a currently being viewed stream (not all of them).